pkg/declared/resources.go

```diff
@@ -68,6 +70,17 @@ func (r *Resources) Update(ctx context.Context, objects []client.Object, commit
 
 	// Record the declared_resources metric, after parsing but before validation.
 	metrics.RecordDeclaredResources(ctx, commit, len(newObjects))
+	if r.previousCommit != commit && r.previousCommit != "" {
+		// Reset the stream value of previous commit to 0 to provide correct input
```

Comment:
Suggested comment:

For Cloud Monitoring, we have configured otel-collector to remove the commit label, to reduce cardinality, but this aggregation uses the max value (b/321875474). So in order for the latest commit to be chosen as the max value, we reset the previous commit value to zero.
TODO: Remove this workaround after migrating to the otel-collector metrics client and switching from gauge to async gauge

Reply:
Done!